docs/proposals/network-policy.md

```diff
@@ -0,0 +1,335 @@
+```
+
+The above structures will be represented in v1beta1 as a json encoded annotation like so:
+
+```yaml
+kind: Namespace
+apiVersion: v1
+metadata:
+  annotations:
+    net.beta.kubernetes.io/network-policy: |
+      {
+        "ingress": {
+          "isolation": "DefaultDeny"
+        }
+      }
+```
+
+### NetworkPolicy Go Definition
+
+For a namespace with ingress isolation, connections to pods in that namespace (from any source) are prevented.  
+The user needs a way to explicitly declare which connections are allowed into pods of that namespace.
+
+This is accomplished through ingress rules on `NetworkPolicy`
+objects (of which there can be multiple in a single namespace).  Pods selected by
+one or more NetworkPolicy objects should allow any incoming connections that match any
+ingress rule on those NetworkPolicy objects, per the network plugin’s capabilities.
+
+NetworkPolicy objects and the above namespace isolation both act on _connections_ rather than individual packets.  That is to say that if traffic from pod A to pod B is allowed by the configured
+policy, then the return packets for that connection from B -> A are also allowed, even if policy is in place that would prevent
+B from initiating a connection to A.  NetworkPolicy objects act on a broad definition of _connection_ which includes both TCP and UDP streams.   If new network policy is applied that would block an existing connection between two endpoints, the enforcer of policy
+should terminate and block the existing connection as soon as can be expected by the implementation.
+
+We propose adding the new NetworkPolicy object to the `extensions/v1beta1` API group for now.
+
+The SIG also considered the following while developing the proposed NetworkPolicy object:
+- A per-pod policy field.  We discounted this in favor of the loose coupling that labels provide, similar to Services.
+- Per-Service policy.  We chose not to attach network policy to services to avoid semantic overloading of a single object, and conflating the existing semantics of load-balancing and service discovery with those of network policy.
+
+```go
+type NetworkPolicy struct {
+	TypeMeta
+	ObjectMeta
+	
+	// Specification of the desired behavior for this NetworkPolicy.
+	Spec NetworkPolicySpec 
+}
+
+type NetworkPolicySpec struct {
+	// Selects the pods to which this NetworkPolicy object applies.  The array of ingress rules 
+	// is applied to any pods selected by this field. Multiple network policies can select the 
+	// same set of pods.  In this case, the ingress rules for each are combined additively.
+	// This field is NOT optional and follows standard unversioned.LabelSelector semantics.  
+	// An empty podSelector matches all pods in this namespace.
+	PodSelector unversioned.LabelSelector `json:"podSelector"`
+	
+	// List of ingress rules to be applied to the selected pods.
+	// Traffic is allowed to a pod if namespace.networkPolicy.ingress.isolation is undefined and cluster policy allows it, 
```

Comment:
The reader has not seen a definition of "cluster policy", and I thought we were calling it "cluster default policy".
The idea of a cluster [default] policy should be defined somewhere.  I think it should be introduced, even if not carefully defined, in the abstract.  I am still a little unclear on how this file becomes what users see, so am not sure exactly what to recommend about where this concept is carefully defined.  At least, it must be properly referenced from the pseudo-code.
I think it really is a *default* policy, because it does not apply if the user chooses to exert policy control.